### PR TITLE
Infra/expandable calendar to function

### PR DESCRIPTION
**Pull request by Inbal-Tish**

### CODE REVIEW

File: src/expandableCalendar/Context/Provider.tsx [hunk 1]
```diff
@@ -34,158 +43,132 @@ interface Props {
...
-    return (
-      <CalendarContext.Provider value={this.getProviderContextValue()}>
-        <View style={[styles.container, this.props.style]}>{this.props.children}</View>
-        {this.props.showTodayButton && this.renderTodayButton()}
-      </CalendarContext.Provider>
-    );
-  }
-}
+  return (
+    <CalendarContext.Provider value={getProviderContextValue()}>
```
Comment (ethanshar) [inline src/expandableCalendar/Context/Provider.tsx:156]: By invoking `getProviderContextValue` each render you create a new object and a new ref

File: src/expandableCalendar/Context/Provider.tsx [hunk 2]
```diff
@@ -34,158 +43,132 @@ interface Props {
...
+      <View style={[style.current.contextWrapper, propsStyle]}>{children}</View>
```
Comment (ethanshar) [inline src/expandableCalendar/Context/Provider.tsx:157]: consider memoizing this `[style.current.contextWrapper, propsStyle]`

Comment (ethanshar) [inline src/expandableCalendar/Context/Provider.tsx:157]: consider memoizing this `[style.current.contextWrapper, propsStyle]`

File: src/expandableCalendar/index.tsx [hunk 3]
```diff
@@ -91,515 +96,491 @@ interface State {
...
+        <Calendar
+          testID="calendar"
+          {...others}
+          theme={themeObject}
+          onDayPress={_onDayPress}
+          hideExtraDays
+          renderArrow={_renderArrow}
+        />
+      ) : (
+        <Animated.View ref={wrapper} style={{height: deltaY.current}} {...panResponder.panHandlers}>
```
Comment (ethanshar) [inline src/expandableCalendar/index.tsx:527]: consider memoizing even that it depends on a ref object. 
The object here always recreated

Comment (ethanshar) [inline src/expandableCalendar/index.tsx:527]: consider memoizing even that it depends on a ref object. 
The object here always recreated

**Review by ethanshar — COMMENTED** [review]
Overall look good, wrote a few comments. 
In general, I see there not a few places where you pass style inline and not memoizing, I think the thumb rule of this should be to use memoize for components that are wrappers for other components cause then a re-render of parent will cause a render of the whole tree under it. 

Another thing, I'd consider breaking big components like expandableCalendar (~550 lines) to smaller parts, using hooks for example. It's really difficult to go over the whole component - you can do it on another PR, not necessarily this one.

File: src/expandableCalendar/Context/Provider.tsx (hunk 1 above)
Comment (Inbal-Tish) [reply src/expandableCalendar/Context/Provider.tsx:156]: Moved to useMemo

File: src/expandableCalendar/Context/Provider.tsx (hunk 2 above)
Comment (Inbal-Tish) [reply src/expandableCalendar/Context/Provider.tsx:157]: moved to useMemo

Comment (Inbal-Tish) [reply src/expandableCalendar/Context/Provider.tsx]: duplication I'm guessing...

File: src/expandableCalendar/index.tsx (hunk 3 above)
Comment (Inbal-Tish) [reply src/expandableCalendar/index.tsx]: Moved all array styles to useMemo

File: src/expandableCalendar/Context/Provider.tsx [hunk 4]
```diff
@@ -34,158 +43,136 @@ interface Props {
...
+      animateOpacity(disabled);
     };
 
-    setDisabled(showTodayButton, disabled, this.state.disabled, updateState);
+    setDisabled(showTodayButton, disabled, isDisabled, updateState);
   };
 
-  animateTodayButton(date: string) {
-    const {shouldAnimateTodayButton, getPositionAnimation} = this.presenter;
+  const getProviderContextValue = useMemo(() => {
```
Comment (ethanshar) [inline src/expandableCalendar/Context/Provider.tsx:106]: since the const is not a method anymore, I suggest renaming it

**Review by ethanshar — APPROVED** [review]
Approved, wrote a small comment on the provider value const name 
Consider fixing it before merging

**Comment by Inbal-Tish** [comment]
> getProviderContextValue

Done!

**Pull request merged**